Basic/DyanmicPrefabs/Assets/Scripts/UI/IPMenuUI.cs

```diff
@@ -0,0 +1,150 @@
+using System;
+using System.Text.RegularExpressions;
+using Unity.Netcode;
+using Unity.Netcode.Transports.UTP;
+using UnityEngine;
+using UnityEngine.UIElements;
+
+namespace Game.UI
+{
+    public class IPMenuUI : MonoBehaviour
+    {
+        // UI Documents
+        [SerializeField]
+        UIDocument m_IPMenuUIDocument;
+        [SerializeField]
+        UIDocument m_ConnectionTypeUIDocument;
+
+        // UI Roots
+        VisualElement m_IPMenuUIRoot;
+        VisualElement m_ConnectionTypeUIRoot;
+
+        // UI Elements
+        TextField m_IPInputField;
+        TextField m_PortInputField;
+        Button m_ButtonHost;
+        Button m_ButtonServer;
+        Button m_ButtonClient;
+        Button m_ButtonDisconnect;
+        Label m_ConnectionTypeLabel;
+
+        void Awake()
+        {
+            SetupIPInputUI();
+
+            // register buttons to methods using callbacks for when they're clicked 
+            m_ButtonHost.clickable.clicked += StartHost;
+            m_ButtonServer.clickable.clicked += StartServer;
+            m_ButtonClient.clickable.clicked += StartClient;
+            m_ButtonDisconnect.clickable.clicked += Disconnect;
+        }
+
+        void OnDestroy()
+        {
+            // un-register buttons from methods using callbacks for when they're clicked 
+            m_ButtonHost.clickable.clicked -= StartHost;
+            m_ButtonServer.clickable.clicked -= StartServer;
+            m_ButtonClient.clickable.clicked -= StartClient;
+            m_ButtonDisconnect.clickable.clicked -= Disconnect;
+
+            NetworkManager.Singleton.OnClientDisconnectCallback -= SingletonOnOnClientDisconnectCallback;
+        }
+
+        void Start()
+        {
+            SetUIElementVisibility(m_IPMenuUIRoot, true);
+            SetUIElementVisibility(m_ConnectionTypeUIRoot, false);
+
+            NetworkManager.Singleton.OnClientDisconnectCallback += SingletonOnOnClientDisconnectCallback;
+        }
+
+        void SingletonOnOnClientDisconnectCallback(ulong obj)
+        {
+            // todo: add logic for server to not do it when a different client disconnects
```

Comment:
`if (obj == NetworkManager.Singleton.LocalClientId)` should do the trick, although you might want to rename that parameter to something like `clientId` instead of `obj`

Reply:
Thanks for taking a look at this! As an update after some iterating on implementing the UI, we're going to actually move any logic for networking out of the UI class and keep it as just UI logic hookups! The networking logic will most likely be handled in the AppController class